pkg/registry/apis/folders/legacy_storage.go

```diff
@@ -0,0 +1,260 @@
+package folders
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"github.com/grafana/grafana/pkg/apis/folders/v0alpha1"
+	"github.com/grafana/grafana/pkg/infra/appcontext"
+	"github.com/grafana/grafana/pkg/kinds"
+	"github.com/grafana/grafana/pkg/services/dashboards"
+	"github.com/grafana/grafana/pkg/services/folder"
+	"github.com/grafana/grafana/pkg/services/grafana-apiserver/endpoints/request"
+)
+
+var (
+	_ rest.Scoper               = (*legacyStorage)(nil)
+	_ rest.SingularNameProvider = (*legacyStorage)(nil)
+	_ rest.Getter               = (*legacyStorage)(nil)
+	_ rest.Lister               = (*legacyStorage)(nil)
+	_ rest.Storage              = (*legacyStorage)(nil)
+	_ rest.Creater              = (*legacyStorage)(nil)
+	_ rest.Updater              = (*legacyStorage)(nil)
+	_ rest.GracefulDeleter      = (*legacyStorage)(nil)
+)
+
+type legacyStorage struct {
+	service        folder.Service
+	namespacer     request.NamespaceMapper
+	tableConverter rest.TableConvertor
+}
+
+func (s *legacyStorage) New() runtime.Object {
+	return resourceInfo.NewFunc()
+}
+
+func (s *legacyStorage) Destroy() {}
+
+func (s *legacyStorage) NamespaceScoped() bool {
+	return true // namespace == org
+}
+
+func (s *legacyStorage) GetSingularName() string {
+	return resourceInfo.GetSingularName()
+}
+
+func (s *legacyStorage) NewList() runtime.Object {
+	return resourceInfo.NewListFunc()
+}
+
+func (s *legacyStorage) ConvertToTable(ctx context.Context, object runtime.Object, tableOptions runtime.Object) (*metav1.Table, error) {
+	return s.tableConverter.ConvertToTable(ctx, object, tableOptions)
+}
+
+func (s *legacyStorage) List(ctx context.Context, options *internalversion.ListOptions) (runtime.Object, error) {
+	orgId, err := request.OrgIDForList(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	limit := int64(5000)
+	if options.Limit > 0 {
+		limit = options.Limit
+	}
+
+	user, err := appcontext.User(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO??? can the folder service return all folders?
+	// When nested folders are not enabled, all folders are root folders
+	hits, err := s.service.GetChildren(ctx, &folder.GetChildrenQuery{
+		SignedInUser: user,
+		Limit:        limit,
+		OrgID:        orgId,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	list := &v0alpha1.FolderList{}
+	for _, v := range hits {
+		list.Items = append(list.Items, *convertToK8sResource(v, s.namespacer))
+	}
+	if len(list.Items) == int(limit) {
+		list.Continue = "<more>" // TODO?
```

Comment:
```suggestion
		ist.Continue = base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%d", page+1)))
```

I have tested the pagination and it works if I add the following block:
```
	page := 1
	if options.Continue != "" {
		continueVal, err := base64.StdEncoding.DecodeString(options.Continue)
		if err != nil {
			return nil, err
		}
		page, err = strconv.Atoi(string(continueVal))
		if err != nil {
			return nil, err
		}
	}
```

and then pass the `page` to the `GetChildren()`